core/capabilities/remote/target/client.go

```diff
@@ -52,48 +52,61 @@ func NewClient(remoteCapabilityInfo commoncap.CapabilityInfo, localDonInfo commo
 	}
 }
 
-func (c *client) expireRequests() {
-	c.mutex.Lock()
-	defer c.mutex.Unlock()
-
-	for messageID, req := range c.messageIDToCallerRequest {
-		if req.Expired() {
-			req.Cancel(errors.New("request expired"))
-			delete(c.messageIDToCallerRequest, messageID)
-		}
-	}
-}
-
 func (c *client) Start(ctx context.Context) error {
 	return c.StartOnce(c.Name(), func() error {
 		c.wg.Add(1)
 		go func() {
 			defer c.wg.Done()
-			ticker := time.NewTicker(c.requestTimeout)
-			defer ticker.Stop()
-			c.lggr.Info("TargetClient started")
-			for {
-				select {
-				case <-c.stopCh:
-					return
-				case <-ticker.C:
-					c.expireRequests()
-				}
-			}
+			c.startCheckingForExpiredRequests()
```

Comment:
nit: you can put "defer" inside startCheckingForExpiredRequests and skip the extra "func()" around it.

Reply:
actually this was in intentional as it places all code related to starting go routines and wait groups in the start and close method which I think is better to keep it close together